src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java

```diff
@@ -29,58 +28,39 @@
 
     private static final Pair<Indexable, Persistable> PAIR_OF_NULLS = new Pair<>(null, null);
 
-    private final List<String> columnFamilyNames = Arrays.asList(
-        new String(RocksDB.DEFAULT_COLUMN_FAMILY),
-        "spentAddress",
-        "transaction",
-        "transaction-metadata",
-        "milestone",
-        "stateDiff",
-        "address",
-        "approvee",
-        "bundle",
-        "obsoleteTag",
-        "tag"
-    );
-
     private final List<ColumnFamilyHandle> columnFamilyHandles = new ArrayList<>();
     private final SecureRandom seed = new SecureRandom();
 
     private final String dbPath;
     private final String logPath;
     private final int cacheSize;
-
-    private ColumnFamilyHandle spentAddressHandle;
-    private ColumnFamilyHandle transactionHandle;
-    private ColumnFamilyHandle transactionMetadataHandle;
-    private ColumnFamilyHandle milestoneHandle;
-    private ColumnFamilyHandle stateDiffHandle;
-    private ColumnFamilyHandle addressHandle;
-    private ColumnFamilyHandle approveeHandle;
-    private ColumnFamilyHandle bundleHandle;
-    private ColumnFamilyHandle obsoleteTagHandle;
-    private ColumnFamilyHandle tagHandle;
+    private final Map<String, Class<? extends Persistable>> columnFamilies;
+    private final Pair<String, Class<? extends Persistable>> metadataColumnFamily;
 
     private Map<Class<?>, ColumnFamilyHandle> classTreeMap;
-    private Map<Class<?>, ColumnFamilyHandle> metadataReference;
+    private Map<Class<?>, ColumnFamilyHandle> metadataReference = Collections.emptyMap();
 
     private RocksDB db;
     // DBOptions is only used in initDB(). However, it is closeable - so we keep a reference for shutdown.
     private DBOptions options;
     private BloomFilter bloomFilter;
     private boolean available;
 
-    public RocksDBPersistenceProvider(String dbPath, String logPath, int cacheSize) {
+    public RocksDBPersistenceProvider(String dbPath, String logPath, int cacheSize,
+                                      Map<String, Class<? extends Persistable>> columnFamilies,
```

Comment:
.. Continue from Iota.java: 
And then replace this part by reading all enum fields & class value, which we add to the map